### PR TITLE
PMM-12376 single inventory add endpoint

### PR DESCRIPTION
**Pull request by atymchuk**
PMM-12376

Link to the Feature Build: SUBMODULES-3366

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [x] API Docs updated
- [x] Add API tests (https://pmm.cd.percona.com/job/pmm2-api-tests/14099/)

This is how the deprecation of endpoints looks like:
Screenshot 2023-08-21 at 08 57 52

These are the endpoint params:
Screenshot 2023-08-25 at 16 38 20

### CODE REVIEW

File: api/inventorypb/nodes.proto [hunk 1]
```diff
@@ -98,9 +98,6 @@ message RemoteRDSNode {
   string az = 6;
   // Custom user-assigned labels.
   map<string, string> custom_labels = 7;
-  // TODO https://jira.percona.com/browse/PMM-4314
```
Comment (atymchuk) [inline api/inventorypb/nodes.proto:101]: This ticket is "Done".

File: managed/services/inventory/grpc/nodes_server.go [hunk 2]
```diff
@@ -106,6 +106,10 @@ func (s *nodesServer) GetNode(ctx context.Context, req *inventorypb.GetNodeReque
 	return res, nil
 }
 
+func (s *nodesServer) AddNode(ctx context.Context, req *inventorypb.AddNodeRequest) (*inventorypb.AddNodeResponse, error) {
+	return s.svc.AddNode(ctx, req)
+}
+
```
Comment (artemgavrilov) [inline managed/services/inventory/grpc/nodes_server.go:112]: Minor: I think we can remove this file at all. Some time ago we agreed that these `grpc` packages are redundant.

File: api/inventorypb/nodes.proto [hunk 3]
```diff
@@ -153,6 +150,27 @@ message GetNodeResponse {
...
 
+// Add any Node
+
+message AddNodeRequest {
+  NodeType node_type = 1;
+  optional AddGenericNodeRequest generic = 2;
+  optional AddContainerNodeRequest container = 3;
+  optional AddRemoteNodeRequest remote = 4;
+  optional AddRemoteRDSNodeRequest remote_rds = 5;
+  optional AddRemoteAzureDatabaseNodeRequest remote_azure = 6;
```
Comment (artemgavrilov) [inline api/inventorypb/nodes.proto:161]: Why not `oneof`?

Comment (atymchuk) [reply api/inventorypb/nodes.proto]: Done.

**Comment by atymchuk** [comment]
> Also API tests are missing

Thanks for pointing it out. Done now 🎉

File: managed/services/inventory/grpc/nodes_server.go (hunk 2 above)
Comment (atymchuk) [reply managed/services/inventory/grpc/nodes_server.go:112]: Ok, we should remove all the three in one shot IMO. Will do as a separate effort.

File: api-tests/inventory/nodes_test.go [hunk 4]
```diff
@@ -91,6 +91,78 @@ func TestNodes(t *testing.T) {
...
+				Address:  "10.10.10.2",
+			},
+		})
+		genericNodeID := genericNode.Generic.NodeID
+		require.NotEmpty(t, genericNodeID)
+		t.Cleanup(func() { pmmapitests.RemoveNodes(t, genericNodeID) })
+
+		res, err := client.Default.Nodes.ListNodes(nil)
+		require.NoError(t, err)
+		require.NotZerof(t, len(res.Payload.Generic), "There should be at least one node")
```
Comment (artemgavrilov) [inline api-tests/inventory/nodes_test.go:120]:
```suggestion
		require.NotEmptyf(t, res.Payload.Generic, "There should be at least one node")
```

File: api/inventorypb/nodes.proto [hunk 5]
```diff
@@ -153,6 +150,28 @@ message GetNodeResponse {
...
 
+// Add any Node
+
+message AddNodeRequest {
+  oneof request {
+    AddGenericNodeRequest generic = 2;
+    AddContainerNodeRequest container = 3;
+    AddRemoteNodeRequest remote = 4;
+    AddRemoteRDSNodeRequest remote_rds = 5;
+    AddRemoteAzureDatabaseNodeRequest remote_azure = 6;
```
Comment (artemgavrilov) [inline api/inventorypb/nodes.proto:161]:
```suggestion
    AddGenericNodeRequest generic = 1;
    AddContainerNodeRequest container = 2;
    AddRemoteNodeRequest remote = 3;
    AddRemoteRDSNodeRequest remote_rds = 4;
    AddRemoteAzureDatabaseNodeRequest remote_azure = 5;
```

**Review by artemgavrilov — APPROVED** [review]

File: api/inventorypb/nodes.proto (hunk 5 above)
Comment (atymchuk) [reply api/inventorypb/nodes.proto:161]: Great catch, thanks!

File: api/inventorypb/nodes.proto [hunk 6]
```diff
@@ -153,6 +150,28 @@ message GetNodeResponse {
   }
 }
 
+// Add any Node
+
+message AddNodeRequest {
+  oneof request {
```
Comment (BupycHuk) [inline api/inventorypb/nodes.proto:156]: maybe `node`?

Comment (BupycHuk) [reply api/inventorypb/nodes.proto:156]: What do you think, can we extract common parameters to a separate field?

**Review by BupycHuk — APPROVED** [review]

File: api-tests/inventory/nodes_test.go (hunk 4 above)
Comment (atymchuk) [reply api-tests/inventory/nodes_test.go]: Ok, I'will mass replace those then ))

Comment (atymchuk) [reply api-tests/inventory/nodes_test.go]: Done.

**Pull request merged**